fuse_models/src/imu_2d.cpp

```diff
@@ -105,47 +106,61 @@ void Imu2D::process(const sensor_msgs::Imu::ConstPtr& msg)
   auto transaction = fuse_core::Transaction::make_shared();
   transaction->stamp(msg->header.stamp);
 
+  // Transform to target frame
+  sensor_msgs::Imu imu_transformed;
+  try
+  {
+    tf_buffer_.transform(*msg, imu_transformed, params_.twist_target_frame);
```

Comment:
There are currently multiple target frames supported:
https://github.com/locusrobotics/fuse/blob/devel/fuse_models/include/fuse_models/parameters/imu_2d_params.h#L99-L101

...though I'm not entirely sure why. I would think that both the twist and acceleration would want to use the same target/body frame.

And it's been a long time since I thought about IMUs, but my recollection is that transforming the IMU pose is complicated. I need to review the imu_transformer package in detail. It was not clear to me how the orientation was being transformed.

Reply:
Yes, that's something I wanted to discuss in this PR and I forgot to mention it.

Right now there are three different target frames:
* `acceleration_target_frame`
* `orientation_target_frame`
* `twist_target_frame`

However, the implementation in [imu_transformer](https://github.com/ros-perception/imu_pipeline/blob/ec36441c5c0cf1e63c12731a3a59185d29a00979/imu_transformer/include/imu_transformer/tf2_sensor_msgs.h#L80-L121) transforms everything in the IMU all at once with a single target frame.

I wonder if it really makes sense to have multiple target frames. :thinking: 

In that case, we could call the `imu_transformer` code three times and only take the components we're interested in, with each target frame. However, in that case, it'd be more efficient, and I believe still clean, to _steal_ some of the code in [imu_transformer](https://github.com/ros-perception/imu_pipeline/blob/ec36441c5c0cf1e63c12731a3a59185d29a00979/imu_transformer/include/imu_transformer/tf2_sensor_msgs.h#L80-L121) to perform the same transformation for each component separately.

I believe the twist is already done correctly, and probably the acceleration as well. The orientation is definitely incorrect.

Just let me know whether you want to keep the three target frames or not, and I'll make the changes mentioned above?

If we remove the existing target frames and create a singal `target_frame`, then we need some deprecation warnings, and set the new `target_frame` with them. In that case, if the different target frames aren't the same, I'm not sure which one should be used. I've been using `twist_target_frame` in this initial implementation.